apps/system/utils/security_level_cmd.c

```diff
@@ -0,0 +1,97 @@
+/****************************************************************************
+ *
+ * Copyright 2023 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/ss_info.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/types.h>
+#include <apps/shell/tash.h>
+#include <security/security_ss.h>
+
+/****************************************************************************
+ * Preprocessor Definitions
+ ****************************************************************************/
+
+/********************************************************************************
+ * Private Function Prototypes
+ ********************************************************************************/
+
+static void tash_set_security_level(int argc, char **args)
+{
+	int security_level;
+	char ss_path[7] = { 0, };
+        security_handle hnd;
+	security_data ss_data = {NULL, 0};
+	security_error ret;
+
+	/* Change security_level according to user input */
+	ss_data.length = 2;
+	if (!strncmp(args[1], "high", 5)) {
+		ss_data.data = "1";
+		security_level = 1;
+	} else if (!strncmp(args[1], "low", 4)) {
+		ss_data.data = "0";
+		security_level = 0;
+	} else {
+		printf("Invalid argument\n");
+		return;
+	}
+
+        /* init */
+	ret = security_init(&hnd);
+        if (ret != SECURITY_OK) {
+		printf("[security_init] error ret : %d\n", ret);
+		return;
+	}
+
+	/* Convert secure storage slot index to string path */
+	snprintf(ss_path, 7, "ss/%d", SECURITY_LEVEL_SS_SLOT_INDEX);
+
+        /* Write 'security_level' variable to secure storage  */
+	ret = ss_write_secure_storage(hnd, ss_path, 0, &ss_data);
+        if (ret != SECURITY_OK) {
+		printf("[ss_write_secure_storage] error ret : %d\n", ret);
+		return;
+	}
+
+	/* Read 'security_level' variable from secure storage  */
+	ret = ss_read_secure_storage(hnd, ss_path, 0, &ss_data);
+        if (ret != SECURITY_OK) {
+		printf("[ss_read_secure_storage] error ret : %d\n", ret);
+		return;
+	}
+
+	if (security_level == 1) {
+		if (!strncmp(ss_data.data, "1", 2)) {
+			printf("security level : high\n");
+		} else {
+			printf("Failed to set security_level\n");
+		}
+	} else {
+		if (!strncmp(ss_data.data, "0", 2)) {
+			printf("security level : low\n");
+		} else {
+			printf("Failed to set security_level\n");
```

Comment:
I don't know why this is necessary. You already check the return value of the write function. Even it returns OK, it could be unwritten?? That's why you read again to confirm? If yes, it's a serious problem. We never trust the API.